docs(theming): add theming reference to tutorialkit.dev (#97)
stackblitz/tutorialkit: docs/tutorialkit.dev/src/content/docs/reference/theming.mdx

---
title: Theming
description: 'How to theme TutorialKit'
---

> This page describes how to customize the appearance of TutorialKit.
> 

Reviewer: What about:
```suggestion
TutorialKit allows you to customize its appearance to match your brand or preferences. This guide explains how to modify various elements of the interface using CSS variables and file replacements.
```

```css
:root[data-theme='light'] {
  /* Light mode colors */
}

:root[data-theme='dark'] {
  /* Dark mode colors */
}
```

Withing these two root selectors, you can define any of the CSS variables mentioned in the following sections.

These tokens can be used to style the colors of the tutorial on a high level. For more detailed customization, you can use the tokens for the individual elements.

| Token                                   | Description                                  |
| --------------------------------------- | -------------------------------------------- |
| `--tk-elements-app-textColor`           | The text color of the TutorialKit app.       |
| `--tk-elements-app-backgroundColor`     | The background color of the TutorialKit app. |
| `--tk-elements-app-borderColor`         | The border color of the TutorialKit app.     |
| `--tk-elements-app-linkColor`           | The link color of the TutorialKit app.       |

> 

Reviewer: Nit: Remove extra newline here.

Author: It's personal preference, and also something that is done often in README files, to add 2 newlines in between the main headers to separate it visually while working on the content. But I'll remove them.

Reviewer: Oh no, no strong preference. I am personally just not a big fan of extra newlines but if we want them to more visually separate them then let's keep them. It's fine but we should be consistent then that's all 👍

Author: I'll check the other documents. If we have it, then I'll leave it in, otherwise I'll remove them.

### Content

The content refers to the main part of the lesson that contains the text and images.

![Content](./images/theming-content.png)

| Token                                    | Description                               |
| ---------------------------------------- | ----------------------------------------- |
| `--tk-elements-content-textColor`        | The text color of the content.            |
| `--tk-elements-content-headingTextColor` | The color of the headings in the content. |

> 

Reviewer: Remove extra newline.

#### Success

| Token                                                | Description                                                     |
| ---------------------------------------------------- | --------------------------------------------------------------- |
| `--tk-elements-callouts-success-textColor`           | The text color for the success callout.                         |
| `--tk-elements-callouts-success-titleTextColor`      | The title text color for the success callout.                   |
| `--tk-elements-callouts-success-iconColor`           | The icon color for the success callout.                         |
| `--tk-elements-callouts-success-backgroundColor`     | The background color for the success callout.                   |
| `--tk-elements-callouts-success-codeColor`           | The color for a code snippet in the success callout.            |
| `--tk-elements-callouts-success-codeBackgroundColor` | The background color for a code snippet in the success callout. |
| `--tk-elements-callouts-success-borderColor`         | The border color for the success callout.                       |

> 

Reviewer: Remove extra newline.

| `--tk-elements-status-skipped-textColor`  | The text color for skipped the status.                           |
| `--tk-elements-status-skipped-iconColor`  | The icon color for the skipped status. Default is the text color |

#### Disabled

| Token                                      | Description                                                       |
| ------------------------------------------ | ----------------------------------------------------------------- |
| `--tk-elements-status-disabled-textColor`  | The text color for disabled the status.                           |
| `--tk-elements-status-disabled-iconColor`  | The icon color for the disabled status. Default is the text color |

#### Active

| Token                                      | Description                                                       |
| ------------------------------------------ | ----------------------------------------------------------------- |
| `--tk-elements-status-active-textColor`  | The text color for active the status.                           |
| `--tk-elements-status-active-iconColor`  | The icon color for the active status. Default is the text color |

> 

Reviewer: Remove extra newline.

![Top Bar](./images/theming-top-bar.png)

| Token                                  | Description                          |
| -------------------------------------- | ------------------------------------ |
| `--tk-elements-topBar-backgroundColor` | The background color of the top bar. |

#### Icon Button

An icon button is a button with only an icon and no text. The color theme switcher in the top bar is an icon button.

| Token                                                  | Description                                             |
| ------------------------------------------------------ | ------------------------------------------------------- |
| `--tk-elements-topBar-iconButton-backgroundColor`      | The background color of the icon button.                |
| `--tk-elements-topBar-iconButton-backgroundColorHover` | The background color of the icon button when hovering.  |
| `--tk-elements-topBar-iconButton-iconColor`            | The color of the icon.                                  |
| `--tk-elements-topBar-iconButton-iconColorHover`       | The color of the icon when hovering.                    |

> 

Reviewer: Remove extra newline.

### Panels

> Panels are containers which can be resized and can hold different kind of components. For instanceof, the file tree, or the terminal.

Reviewer: ```suggestion
Panels are containers which can be resized and can hold different kind of components. For instance, the file tree or the terminal.
```

| Token                                      | Description                         |
| ------------------------------------------ | ----------------------------------- |
| `--tk-elements-panel-textColor`            | The text color of the panels.       |
| `--tk-elements-panel-backgroundColor`      | The background color of the panels. |

#### Header

> The panel header is the top part of the panel that contains the title, an icon, and optionally an action button

Reviewer: ```suggestion
The panel header is the top part of the panel that contains the title, an icon, and optionally an action button.
```

| `--tk-elements-panel-header-iconColor`       | The color of the icon in the header. |

##### Button

The button in the header is shown at the right side of the header of the panel.

![Panel Header Button](./images/theming-panel-header-button.png)

| Token                                                   | Description                                       |
| ------------------------------------------------------- | ------------------------------------------------- |
| `--tk-elements-panel-headerButton-textColor`            | The text color of the button.                     |
| `--tk-elements-panel-headerButton-textColorHover`       | The text color of the button when hovering.       |
| `--tk-elements-panel-headerButton-backgroundColor`      | The background color of the button.               |
| `--tk-elements-panel-headerButton-backgroundColorHover` | The background color of the button when hovering. |
| `--tk-elements-panel-headerButton-iconColor`            | The icon color.                                   |
| `--tk-elements-panel-headerButton-iconColorHover`       | The icon color when hovering.                     |

##### Tab

> 

Reviewer: Other sections contain a brief description. Shall we add one here too just for consistency?

| `--tk-elements-panel-headerTab-textColorHover`        | The text color of the tab when hovering        |
| `--tk-elements-panel-headerTab-textColorActive`       | The text color of the active tab.              |
| `--tk-elements-panel-headerTab-backgroundColor`       | The background color of the tab.               |
| `--tk-elements-panel-headerTab-backgroundColorHover`  | The background color of the tab when hovering. |
| `--tk-elements-panel-headerTab-backgroundColorActive` | The background color of the active tab.        |
| `--tk-elements-panel-headerTab-borderColor`           | The border color of the tab.                   |
| `--tk-elements-panel-headerTab-borderColorHover`      | The border color of the tab when hovering      |
| `--tk-elements-panel-headerTab-borderColorActive`     | The border color of the active tab.            |
| `--tk-elements-panel-headerTab-iconColor`             | The icon color of the tab.                     |
| `--tk-elements-panel-headerTab-iconColorHover`        | The icon color of the tab when hovering        |
| `--tk-elements-panel-headerTab-iconColorActive`       | The icon color of the active tab.              |

> 

Reviewer: Remove extra newline.

| `--tk-elements-fileTree-file-iconColorSelected`       | The color of the file icon of the selected item.     |

#### Folder

You can customize the appearance of the folders in the file tree using the following tokens.

| Token                                                | Description                                            |
| ---------------------------------------------------- | ------------------------------------------------------ |
| `--tk-elements-fileTree-folder-textColor`            | The text color of the folder item.                     |
| `--tk-elements-fileTree-folder-textColorHover`       | The text color of the folder item when hovering.       |
| `--tk-elements-fileTree-folder-backgroundColor`      | The background color of the folder item.               |
| `--tk-elements-fileTree-folder-backgroundColorHover` | The background color of the folder item when hovering. |
| `--tk-elements-fileTree-folder-iconColor`            | The color of the folder icon.                          |
| `--tk-elements-fileTree-folder-iconColorHover`       | The color of the folder icon when hovering.            |

> 

Reviewer: Remove extra newline.

![Navigation Card](./images/theming-navcard.png)

| Token                                        | Description                                                 |
| -------------------------------------------- | ----------------------------------------------------------- |
| `--tk-elements-navCard-textColor`            | The text color for the navigation card.                     |
| `--tk-elements-navCard-textColorHover`       | The text color for the navigation card when hovering.       |
| `--tk-elements-navCard-backgroundColor`      | The background color for the navigation card.               |
| `--tk-elements-navCard-backgroundColorHover` | The background color for the navigation card when hovering. |
| `--tk-elements-navCard-borderColor`          | The border color for the navigation card.                   |
| `--tk-elements-navCard-borderColorHover`     | The border color for the navigation card when hovering.     |
| `--tk-elements-navCard-iconColor`            | The icon color for the navigation card.                     |
| `--tk-elements-navCard-iconColorHover`       | The icon color for the navigation card when hovering.       |

> 

Reviewer: Remove extra newline.

| `--tk-elements-breadcrumbs-dropdown-accordionIconColor`            | Icon color for the collapsable or expandable title.              |
| `--tk-elements-breadcrumbs-dropdown-accordionIconColorSelected`    | Icon colorfor the selected collapsable or expandable title.      |
| `--tk-elements-breadcrumbs-dropdown-accordionIconColorHover`       | Icon color when hovering over a collapsable or expandable title. |
| `--tk-elements-breadcrumbs-dropdown-lessonBackgroundColor`         | Background color for the lesson.                                 |
| `--tk-elements-breadcrumbs-dropdown-lessonBackgroundColorSelected` | Background color for the selected lesson.                        |
| `--tk-elements-breadcrumbs-dropdown-lessonTextColor`               | Text color for the lesson.                                       |
| `--tk-elements-breadcrumbs-dropdown-lessonTextColorSelected`       | Text color for the selected lesson.                              |
| `--tk-elements-breadcrumbs-dropdown-lessonTextColorHover`          | Text color when hovering over a lesson.                          |

> 

Reviewer: Remove extra newline.

### Previews

The previews can show one or multiple websites.

![Previews](./images/theming-previews.png)

| Token                                | Description                                         |
| ------------------------------------ | --------------------------------------------------- |
| `--tk-elements-previews-borderColor` | The color of the borders between multiple previews. |

> 

Reviewer: Remove extra newline.
